### PR TITLE
[receiver/azureeventhub] Add new component, Azure Event Hub Receiver

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Structure of a new receiver for Azure event hub.

**Link to tracking Issue:**
#12786 

**Testing:**
N/A

**Documentation:**
README.md

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
FYI only: Have not looked at the code, but consider to use same pattern as kafka receiver that allows plugin different "Unmarshaller" (otlp, etc.)

**Comment by atoulme** [comment]
yes, this is just the structure, the impl is not set yet. Good call on the unmarshaller.

**Comment by atoulme** [comment]
@djaglowski I am having issues with flaky tests, please review as I try one more time to get a green build...

File: receiver/azureeventhubreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Azure Event Hub Receiver
+
+| Status                   |                  |
+| ------------------------ |------------------|
+| Stability                | [alpha]          |
```
Comment (djaglowski) [inline receiver/azureeventhubreceiver/README.md:5]: This should be [`in development`](https://github.com/open-telemetry/opentelemetry-collector#in-development).

**Review by djaglowski — COMMENTED** [review]
LGTM as long as stability is changed to "in development".

I haven't seen much test flakiness lately. If you can link me to any failures, I'd like to at least capture them.

File: receiver/azureeventhubreceiver/README.md (hunk 1 above)
Comment (atoulme) [reply receiver/azureeventhubreceiver/README.md]: thanks, fixed

**Review by djaglowski — APPROVED** [review]

**Pull request merged**